cryptomatte/cryptomatte.h

```diff
@@ -490,7 +490,7 @@ inline void offset_name(const AtShaderGlobals* sg, const AtNode* node, const int
                         char obj_name_out[MAX_STRING_LENGTH]) {
     if (offset) {
         char offset_num_str[12];
-        sprintf(offset_num_str, "_%d", offset);
+        snprintf(offset_num_str, 12, "_%d", offset);
```

Comment:
Hardcoded values could get out of sync in the future. At compile time the compiler will replace the sizeof with 12 or whatever new value this buffer might have.
```suggestion
        snprintf(offset_num_str, sizeof(offset_num_str), "_%d", offset);
```

Reply:
I agree that hardcoded value can get out of sync. Generally I would use a const/constexpr value for the buffer size, simply because I have seen mistake where the code changes from
```
char offset [3];
snprintf(offset, sizeof(offset), "%d", value);
```
to 
```
char *offset = new char[x];
snprintf(offset, sizeof(offset), "%d", value);
```
sizeof(offset) will just become 8 and you have a potential buffer overflow. This is common, specially if the developer is not sure about the nuance between char[] and char * or hasn't spotted the sizeof.

So in my experience having the size of the buffer explicitly defined is more future proof, like:
```
constexpr size_t offset_buffer_size = 3;
char *offset = new char[offset_buffer_size];
snprintf(offset, offset_buffer_size, "%d", value);
....
char offset2[offset_buffer_size];
snprintf(offset2, offset_buffer_size, "%d", value);

```
I feel that in the current code, the locality between the buffer declaration and the snprintf is visually helping to avoid mistake, better than sizeof, what do you think ? But I can change it, no worries.

Reply:
Fair enough, then how about using `std::array` for the buffer? That compiles down to the same assembly code but it's even more explicit about what is going on and much less likely that someone would break it accidentally.

Reply:
`std::array` would work as well !